streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -804,21 +804,26 @@ long runOnce(final long recordsProcessedBeforeCommit) {
             // try to fetch some records with zero poll millis
             // to unblock the restoration as soon as possible
             records = pollRequests(Duration.ZERO);
+        } else if (state == State.PARTITIONS_REVOKED) {
+            // try to fetch some records with normal poll time
+            // in order to wait long enough to get the join response
+            records = pollRequests(pollTime);
+        } else if (state == State.RUNNING) {
+            // try to fetch some records with normal poll time
+            // in order to get long polling
+            records = pollRequests(pollTime);
```

Comment:
nit: should we unify `PARTITONS_REVOKED` and ` RUNNING`?

Reply:
I've thought about that, and I think moving forward we may want still use different poll time for RUNNING and PARTITION_REVOKED as well, since for former we should use normal long polling, while for latter there may be some optimization we can do regarding the consumer internals. But those ideas need to be validated via some profiling. I'm intentionally leaving them separated for now.